something went wrong in my_code()
throw an exception forcibly
